homeassistant/components/octoprint/__init__.py

```diff
@@ -134,6 +152,76 @@ def device_discovered(service, info):
                       config)
         success = True
 
+    def handle_cancel_job(call):
+        """Abort the current job."""
+        octoprint_api.post('job', "{\"command\": \"cancel\"}")
```

Comment:
Please create a third party lib to control Octoprint. We shouldn't be including protocol specific information. 

Also, you might in your lib prefer to use `json.dumps({"command": "cancel"})` instead of writing raw JSON.